### ISSUE TITLE
Errors remain after being fixed, until file is reopened

### ISSUE THREAD

**Issue by markedwards** [microsoft/vscode-mypy#130]
(using pre-release v2023.3.12361006, because the current release doesn't work at all due to a stray `--follow-imports=silent` option)

Steps to reproduce:

1. Create a file with:

```python
def temp() -> str:
    return "0"
```

2. Observe there are no type errors (as expected).
3. Change line 2 to `return 0`
4. Observe there is now a type error complaining about the incorrect int return value.
5. Fix the bug by updating the code again to `return "0"`
6. Observe that the error is still displayed, even though there is no error now in the Mypy Type Checker output tab.
7. Close file and reopen, observe that the error is now gone.

**Comment by karthiknadig** [comment]
This is https://github.com/microsoft/vscode-mypy/issues/88 which is caused by https://github.com/python/mypy/issues/15677

We plan us switching to `mypy` via a setting to handle these scenarios.

**Comment by markedwards** [comment]
Okay, just to be clear, the Mypy Type Checker output does not agree with what is displayed in the editor.  And I cannot reproduce this my running `dmypy` directly.

**Comment by karthiknadig** [comment]
Ahh! the stary `silent` issue. I am working on removing the legacy settings support.

**Comment by markedwards** [comment]
I think the behavior I'm describing above is different from #88, and not related to that mypy issue.  In my case, mypy and dmypy and the output tab all say the correct thing.  The only issue is that the display in the VSCode editor is wrong.  And it is fixed by simply closing and opening the file in the editor.  No need to reload the VSCode window.

**Comment by phillies** [comment]
I can confirm this issue. Reopening the file or restarting the server fix the issue. 
Log is fine and no error is shown there, but the error (red lines + mouseover) are still in the editor.

Version: 1.81.1 (user setup)
Commit: 6c3e3dba23e8fadc360aed75ce363ba185c49794
Date: 2023-08-09T22:22:42.175Z
Electron: 22.3.18
ElectronBuildId: 22689846
Chromium: 108.0.5359.215
Node.js: 16.17.1
V8: 10.8.168.25-electron.0
OS: Windows_NT x64 10.0.19045
mypy extension v2023.3.12421009
mypy 1.4.1 (same behavior when using bundled mypy)

**Comment by prokie** [comment]
I can confirm this issue as well. I am using the pre-release version of Mypy extension and tried both mypy 1.5.0 and 1.5.1.

**Comment by martinkozle** [comment]
This issue has been pretty annoying for the past few weeks. Having to run `Mypy: Restart Server` constantly to get rid of the false errors.
